yb-voyager/cmd/live_migration_test.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	reporter "github.com/yugabyte/yb-voyager/yb-voyager/src/reporter/stats"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils/sqlname"
+)
+
+type mockYugabyteDB struct {
+	tgtdb.TargetYugabyteDB // to satisfy interface
+}
+
+func (myb *mockYugabyteDB) ExecuteBatch(migrationUUID uuid.UUID, batch *tgtdb.EventBatch) error {
+	return nil
+}
+
+func TestProcessEventsBasic(t *testing.T) {
+	evChan := make(chan *tgtdb.Event, EVENT_CHANNEL_SIZE)
+	lastAppliedVsn := int64(0)
+	doneChan := make(chan bool, 1)
+	statsReporter := &reporter.StreamImportStatsReporter{}
+	state := NewImportDataState(exportDir)
+	tdb = &mockYugabyteDB{}
+	conflictDetectionCache = NewConflictDetectionCache(utils.NewStructMap[sqlname.NameTuple, []string](), []chan *tgtdb.Event{evChan}, "postgresql")
+
+	oname := sqlname.NewObjectName("yugabytedb", "public", "public", "users")
+	evChan <- &tgtdb.Event{
+		Vsn: 1,
+		Op:  "c",
+		TableNameTup: sqlname.NameTuple{
+			CurrentName: oname,
+			TargetName:  oname,
+		},
+		ExporterRole: SOURCE_DB_EXPORTER_ROLE,
+	}
+	evChan <- END_OF_QUEUE_SEGMENT_EVENT
+	processEvents(1, evChan, lastAppliedVsn, doneChan, statsReporter, state)
+}
+
+func TestProcessEventsRemovesEventFromConflicDetectionCache(t *testing.T) {
+	evChan := make(chan *tgtdb.Event, EVENT_CHANNEL_SIZE)
+	lastAppliedVsn := int64(0)
+	doneChan := make(chan bool, 1)
+	statsReporter := &reporter.StreamImportStatsReporter{}
+	state := NewImportDataState(exportDir)
+	tdb = &mockYugabyteDB{}
+	conflictDetectionCache = NewConflictDetectionCache(utils.NewStructMap[sqlname.NameTuple, []string](), []chan *tgtdb.Event{evChan}, "postgresql")
+
+	oname := sqlname.NewObjectName("yugabytedb", "public", "public", "users")
+	e := &tgtdb.Event{
+		Vsn: 1,
+		Op:  "c",
+		TableNameTup: sqlname.NameTuple{
+			CurrentName: oname,
+			TargetName:  oname,
+		},
+		ExporterRole: SOURCE_DB_EXPORTER_ROLE,
+	}
```

Comment:
just thinking out loud, can we some common abstracted code to generate object names/events in the code so that everyone can use that for adding tests, without thinking much about the sqlname or nametuple packages.

Reply:
Definitely makes sense, and something to keep in mind. I think these patterns will unfold as we write more and more unit test cases.